Vulkan compile errors on mingw

**Godot version:**

1037fa4

**OS/device including version:**
compiling on docker / fedora 30

**Issue description:**

Is WinSock2 spelt wrong?

```bash
In file included from thirdparty/vulkan/loader/dev_ext_trampoline.c:22:
thirdparty/vulkan/loader/vk_loader_platform.h:28:10: fatal error: WinSock2.h: No such file or directory
 #include <WinSock2.h>
          ^~~~~~~~~~~~
compilation terminated.
scons: *** [thirdparty/vulkan/loader/dev_ext_trampoline.windows.opt.tools.64.o] Error 1
In file included from thirdparty/vulkan/loader/loader.h:33,
                 from thirdparty/vulkan/loader/asm_offset.c:24:
thirdparty/vulkan/loader/vk_loader_platform.h:28:10: fatal error: WinSock2.h: No such file or directory
 #include <WinSock2.h>
          ^~~~~~~~~~~~
compilation terminated.
scons: *** [thirdparty/vulkan/loader/asm_offset.windows.opt.tools.64.o] Error 1
drivers/wasapi/audio_driver_wasapi.cpp: In member function 'virtual ULONG CMMNotificationClient::Release()':
drivers/wasapi/audio_driver_wasapi.cpp:93:11: warning: deleting object of polymorphic class type 'CMMNotificationClient' which has non-virtual destructor might cause undefined behavior [-Wdelete-non-virtual-dtor]
    delete this;
```

**Steps to reproduce:**

```scons platform=windows target=release_debug use_static_cpp=yes -j`nproc` use_lto=no use_llvm=yes gdnative_wrapper=yes deprecated=no```

--- a/thirdparty/vulkan/loader/loader.c
+++ b/thirdparty/vulkan/loader/loader.c
@@ -60,9 +60,9 @@
 #include "murmurhash.h"
 
 #if defined(_WIN32)
-#include <Cfgmgr32.h>
+#include <cfgmgr32.h>
 #include <initguid.h>
-#include <Devpkey.h>
+#include <devpkey.h>
 #endif
 
 // This is a CMake generated file with #defines for any functions/includes

--- a/thirdparty/vulkan/loader/vk_loader_platform.h
+++ b/thirdparty/vulkan/loader/vk_loader_platform.h
@@ -25,7 +25,7 @@
 
 #if defined(_WIN32)
 // WinSock2.h must be included *BEFORE* windows.h
-#include <WinSock2.h>
+#include <winsock2.h>
 #endif  // _WIN32
 
 #include "vulkan/vk_platform.h"

